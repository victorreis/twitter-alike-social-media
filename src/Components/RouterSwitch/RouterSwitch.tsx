import { createElement } from 'react';
import { Route, Routes } from 'react-router-dom';

import { TestProps } from '../../Config/Tests/Test.types';
import { Home } from '../../Pages/Home';
import { routes } from '../../Routes/Routes';
import {
  RouterSwitchProps,
  DefaultRouterSwitchProps,
} from './RouterSwitch.types';

export const routerSwitchDefaults: Required<DefaultRouterSwitchProps> &
  Required<TestProps> = {
  testID: 'RouterSwitch',
};

export const RouterSwitch: React.FC<RouterSwitchProps> = (
  props
): JSX.Element => {
  const { testID = routerSwitchDefaults.testID } = props;

  const renderRoutes = () => {
    return routes.map(({ key, element, path }) => (
      <Route key={key} element={createElement(element)} path={path} />
    ));
  };

  return (
    <main data-testid={testID}>
      <Routes>
        <Route element={<Home />} path="/" />
        {renderRoutes()}
        <Route element={<Home />} path="*" />
      </Routes>
    </main>
  );
};
