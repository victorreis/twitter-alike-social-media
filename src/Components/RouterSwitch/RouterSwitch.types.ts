import { TestProps } from '../../Config/Tests/Test.types';

export interface RequiredRouterSwitchProps {}

export interface DefaultRouterSwitchProps {}

export interface OptionalRouterSwitchProps {}

export type RouterSwitchProps = RequiredRouterSwitchProps &
  DefaultRouterSwitchProps &
  OptionalRouterSwitchProps &
  TestProps;

export type RouterSwitchStyleProps = Required<DefaultRouterSwitchProps>;
