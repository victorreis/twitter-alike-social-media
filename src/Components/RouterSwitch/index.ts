export * from './RouterSwitch.types';
export * from './RouterSwitch';
