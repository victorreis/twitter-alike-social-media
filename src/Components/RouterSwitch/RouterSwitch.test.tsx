import {
  renderJestDomCreator,
  renderRTRCreator,
  screen,
} from '../../Config/Tests/GlobalSetup.config';
import { RouterSwitch, routerSwitchDefaults } from './RouterSwitch';
import {
  RequiredRouterSwitchProps,
  RouterSwitchProps,
} from './RouterSwitch.types';

describe('routerSwitch component tests', () => {
  const requiredProps: RequiredRouterSwitchProps = {};

  const setup = (props?: RouterSwitchProps) => {
    const renderRTR = () =>
      renderRTRCreator<RouterSwitchProps>(RouterSwitch, {
        ...requiredProps,
        ...props,
      });
    const renderJestDom = () =>
      renderJestDomCreator<RouterSwitchProps>(RouterSwitch, {
        ...requiredProps,
        ...props,
      });

    return { renderRTR, renderJestDom };
  };

  describe('behavior tests', () => {
    it(`should render the component`, () => {
      expect.assertions(1);
      setup().renderJestDom();
      const testInstance = screen.getByTestId(routerSwitchDefaults.testID);

      expect(testInstance).toBeTruthy();
    });
  });

  describe('style tests', () => {});

  describe('snapshot tests', () => {
    it(`should render correctly`, () => {
      expect.assertions(1);
      const generatedJson = setup().renderRTR().toJSON();

      expect(generatedJson).toMatchSnapshot();
    });
  });
});
