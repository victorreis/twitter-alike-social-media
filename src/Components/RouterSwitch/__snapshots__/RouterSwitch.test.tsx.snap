// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routerSwitch component tests snapshot tests should render correctly 1`] = `
<main
  data-testid="RouterSwitch"
>
  <div
    className="ContainerScroll-twitter-alike-social-media__sc-q0f03s-0 PageContainer-twitter-alike-social-media__sc-q0f03s-2 bzejee flxQWw"
    data-testid="Home"
  >
    <nav>
      <div
        className="HomeVerticalMenu-twitter-alike-social-media__sc-yzia5e-2 ijnLPh"
      >
        <a
          aria-current="page"
          className="Logo-twitter-alike-social-media__sc-yzia5e-1 byvFC active"
          href="/"
          onClick={[Function]}
        >
          <span>
            <span
              className="larger"
            >
              P
            </span>
            osterr
          </span>
        </a>
        <a
          aria-current="page"
          className="MenuItemContainer-twitter-alike-social-media__sc-87xn0-0 hUZhVl active"
          data-testid="MenuItem"
          href="/"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M946.5 505L534.6 93.4a31.93 31.93 0 0 0-45.2 0L77.5 505c-12 12-18.8 28.3-18.8 45.3 0 35.3 28.7 64 64 64h43.4V908c0 17.7 14.3 32 32 32H448V716h112v224h265.9c17.7 0 32-14.3 32-32V614.3h43.4c17 0 33.3-6.7 45.3-18.8 24.9-25 24.9-65.5-.1-90.5z"
            />
          </svg>
           Home
        </a>
        <a
          aria-current="page"
          className="MenuItemContainer-twitter-alike-social-media__sc-87xn0-0 hUZhVl active"
          data-testid="MenuItem"
          href="/"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 496 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 96c48.6 0 88 39.4 88 88s-39.4 88-88 88-88-39.4-88-88 39.4-88 88-88zm0 344c-58.7 0-111.3-26.6-146.5-68.2 18.8-35.4 55.6-59.8 98.5-59.8 2.4 0 4.8.4 7.1 1.1 13 4.2 26.6 6.9 40.9 6.9 14.3 0 28-2.7 40.9-6.9 2.3-.7 4.7-1.1 7.1-1.1 42.9 0 79.7 24.4 98.5 59.8C359.3 421.4 306.7 448 248 448z"
            />
          </svg>
          Profile
        </a>
        <button
          className="ButtonContainer-twitter-alike-social-media__sc-1bqjegs-0 iHxNIv"
          data-testid="Button"
          disabled={false}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          size="LG"
          type="button"
        >
          POST
        </button>
      </div>
    </nav>
    <div
      className="HomeFeedContainer-twitter-alike-social-media__sc-yzia5e-3 hruGor"
    >
      <div
        className="FixedBarContainer-twitter-alike-social-media__sc-1hwkltx-0 eprasI"
        data-testid="FixedBar"
      >
        <span
          className="TypographyContainer-twitter-alike-social-media__sc-1rk4wyq-0 llowaZ"
          data-testid="Typography"
        >
          Home
        </span>
        <div
          className="TextToggleContainer-twitter-alike-social-media__sc-oz9uiz-0 isqEII"
          data-testid="TextToggle"
        >
          <button
            className="ButtonContainer-twitter-alike-social-media__sc-1bqjegs-0 dLbeGW TextToggleActiveButton-twitter-alike-social-media__sc-oz9uiz-1 jwXQve"
            data-testid="Button"
            disabled={false}
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            size="MD"
            type="button"
          >
            ALL
          </button>
          <button
            className="ButtonContainer-twitter-alike-social-media__sc-1bqjegs-0 dLbeGW TextToggleActiveButton-twitter-alike-social-media__sc-oz9uiz-1 TextToggleButton-twitter-alike-social-media__sc-oz9uiz-2 jwXQve eUAzmR"
            data-testid="Button"
            disabled={false}
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            size="MD"
            type="button"
          >
            FOLLOWING
          </button>
        </div>
      </div>
      <div
        className="PostCreatorContainer-twitter-alike-social-media__sc-r64v60-0 jgVgol"
        data-testid="PostCreator"
      >
        <div
          className="PostCreatorHeader-twitter-alike-social-media__sc-r64v60-1 jPmJar"
        >
          <div
            className="AvatarContainer-twitter-alike-social-media__sc-ne4tqs-0 bMMLCN"
            data-testid="Avatar"
            onClick={[Function]}
            size="MD"
          >
            <img
              alt="undefined's avatar photo."
              className="AvatarImage-twitter-alike-social-media__sc-ne4tqs-1 cdAMXQ"
              data-testid="Avatar_image"
              onError={[Function]}
              size="MD"
              src="undefined"
              title="undefined's avatar photo."
            />
          </div>
          <textarea
            className="StyledTextarea-twitter-alike-social-media__sc-12f32i7-0 bivTbY"
            data-testid="Textarea"
            maxLength={777}
            onChange={[Function]}
            placeholder="What is happening?"
            rows={4}
            value=""
          />
        </div>
        <div
          className="PostCreatorFooter-twitter-alike-social-media__sc-r64v60-2 guOzgL"
        >
          <span
            className="TypographyContainer-twitter-alike-social-media__sc-1rk4wyq-0 llowaZ"
            data-testid="Typography"
          >
            0
             / 
            777
             
          </span>
          <button
            className="ButtonContainer-twitter-alike-social-media__sc-1bqjegs-0 iHxNIv"
            data-testid="Button"
            disabled={false}
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            size="LG"
            type="button"
          >
            POST
          </button>
        </div>
      </div>
    </div>
  </div>
</main>
`;
